In [1]:
#import libraries 
from shapely.geometry import Point
import pandas as pd
import geopandas as gpd
import matplotlib.pyplot as plt
import folium
import datetime as dt

In [2]:
july_scooter = pd.read_csv('data/july.csv')
july_scooter.head()

pubdatetime   latitude  longitude         sumdid sumdtype  \
0  2019-07-01 00:00:33.550000  36.156678 -86.809004  Powered635135  Powered   
1  2019-07-01 00:00:34.973000  36.145674 -86.794138  Powered790946  Powered   
2  2019-07-01 00:00:41.183000  36.179319 -86.751538  Powered570380  Powered   
3  2019-07-01 00:00:41.620000  36.152111 -86.803821  Powered240631  Powered   
4  2019-07-01 00:00:45.087000  36.149355 -86.797550  Powered970404  Powered   

   chargelevel sumdgroup  costpermin companyname  
0         22.0   scooter        0.15        Lyft  
1         33.0   scooter        0.15        Lyft  
2         76.0   scooter        0.15        Lyft  
3         43.0   scooter        0.15        Lyft  
4         52.0   scooter        0.15        Lyft

In [3]:
july_scooter.sumdgroup.value_counts()

scooter    20193663
Scooter     4872861
bicycle        8921
Name: sumdgroup, dtype: int64

In [4]:
july_scooter.companyname.unique()

array(['Lyft', 'Bird', 'Spin', 'Bolt', 'Jump', 'Lime', 'Gotcha'],
      dtype=object)

In [5]:
july_scooter.sumdid.value_counts()

StandardNW5HJFO4R32LY                          8921
PoweredK7VBEW7S2VHPG                           8915
PoweredSMYDSCUCCKE3D                           8910
PoweredCPBMGOVZOHE4Z                           8909
Powered9edd61b3-3f6e-5b4a-b1ec-dd5b6937f65f    8901
                                               ... 
PoweredBHQPF4NGORWCW                              1
Powered6TPOZJO4U6WGE                              1
PoweredC7EZRZTYAREOS                              1
Powered4314410                                    1
Powered6EPAKQB4QHWJF                              1
Name: sumdid, Length: 6788, dtype: int64

In [6]:
july_scooter.costpermin.value_counts()

0.15    12569385
0.06    10367575
0.23     1750583
0.30      378981
0.05        8921
Name: costpermin, dtype: int64

In [7]:
june_scooter = pd.read_csv('data/june.csv')
june_scooter.head()

pubdatetime  latitude  longitude  \
0  2019-06-01 00:00:12   36.1202   -86.7534   
1  2019-06-01 00:00:12   36.1630   -86.7765   
2  2019-06-01 00:00:12   36.1202   -86.7533   
3  2019-06-01 00:00:12   36.1201   -86.7530   
4  2019-06-01 00:00:12   36.1622   -86.7806   

                                        sumdid sumdtype  chargelevel  \
0  Powered93627c35-0f62-5b81-a78d-75a4a92ecf47  Powered         90.0   
1  Powered17715097-e8a0-5494-a5ab-9b625796607d  Powered         63.0   
2  Powerede5cb95ae-b091-5a93-86fa-ededd946d0d7  Powered         77.0   
3  Powered71fa5e4f-1e17-54c4-936d-330df38cc2fa  Powered          0.0   
4  Poweredfa549dd6-40bb-5757-ac87-2c2528f2bc68  Powered          2.0   

  sumdgroup  costpermin companyname  
0   scooter        0.06        Jump  
1   scooter        0.06        Jump  
2   scooter        0.06        Jump  
3   scooter        0.06        Jump  
4   scooter        0.06        Jump

In [8]:
june_scooter.sumdgroup.value_counts()

scooter    21934928
Scooter     6102480
bicycle        8687
Name: sumdgroup, dtype: int64

In [9]:
june_scooter.companyname.unique()

array(['Jump', 'Bird', 'Bolt', 'Gotcha', 'Spin', 'Lime', 'Lyft'],
      dtype=object)

In [10]:
june_scooter.costpermin.value_counts()

0.15    17046428
0.06     8482893
0.23     2508140
0.05        8634
Name: costpermin, dtype: int64

In [11]:
june_scooter.sumdid.value_counts()

StandardNW5HJFO4R32LY                          8634
PoweredUNORTWSTRM4AM                           8631
PoweredBV42LPYJGVBL4                           8630
Powered6d713a97-9fa3-5af5-ab16-f0a1da6d88f1    8628
Poweredb352730c-ecb6-57ad-a491-f8819be313b8    8628
                                               ... 
PoweredXCPDB                                      1
Powered255293                                     1
PoweredNIBNX6HK6AP5L                              1
Powered6668448                                    1
PoweredUKDFR5O5HBMBO                              1
Name: sumdid, Length: 7828, dtype: int64

In [12]:
may_scooter  = pd.read_csv('data/may.csv')
may_scooter.head()

pubdatetime   latitude  longitude        sumdid sumdtype  \
0  2019-05-01 00:01:41.247000  36.136822 -86.799877  PoweredLIRL1  Powered   
1  2019-05-01 00:01:41.247000  36.191252 -86.772945  PoweredXWRWC  Powered   
2  2019-05-01 00:01:41.247000  36.144752 -86.806293  PoweredMEJEH  Powered   
3  2019-05-01 00:01:41.247000  36.162056 -86.774688  Powered1A7TC  Powered   
4  2019-05-01 00:01:41.247000  36.150973 -86.783109  Powered2TYEF  Powered   

   chargelevel sumdgroup  costpermin companyname  
0         93.0   scooter         0.0        Bird  
1         35.0   scooter         0.0        Bird  
2         90.0   scooter         0.0        Bird  
3         88.0   scooter         0.0        Bird  
4         98.0   scooter         0.0        Bird

In [13]:
may_scooter.companyname.unique()

array(['Bird', 'Lyft', 'Gotcha', 'Lime', 'Spin', 'Jump', 'Bolt'],
      dtype=object)

In [14]:

may_scooter.costpermin.value_counts()

0.00    10918645
0.15     5577488
0.06     2741578
0.23     1051915
0.05        2877
Name: costpermin, dtype: int64

In [15]:
may_scooter['sumdtype'].value_counts()

Powered     20283582
Standard        8921
Name: sumdtype, dtype: int64

In [16]:
may_scooter['sumdgroup'].value_counts()

scooter    17542872
Scooter     2740710
bicycle        8921
Name: sumdgroup, dtype: int64

In [17]:
# dropping sumdtype
july_scooter_new = july_scooter.drop(columns = 'sumdtype')
july_scooter_new.head()

pubdatetime   latitude  longitude         sumdid  \
0  2019-07-01 00:00:33.550000  36.156678 -86.809004  Powered635135   
1  2019-07-01 00:00:34.973000  36.145674 -86.794138  Powered790946   
2  2019-07-01 00:00:41.183000  36.179319 -86.751538  Powered570380   
3  2019-07-01 00:00:41.620000  36.152111 -86.803821  Powered240631   
4  2019-07-01 00:00:45.087000  36.149355 -86.797550  Powered970404   

   chargelevel sumdgroup  costpermin companyname  
0         22.0   scooter        0.15        Lyft  
1         33.0   scooter        0.15        Lyft  
2         76.0   scooter        0.15        Lyft  
3         43.0   scooter        0.15        Lyft  
4         52.0   scooter        0.15        Lyft

In [18]:
june_scooter_new = june_scooter.drop(columns = 'sumdtype')
june_scooter_new.head()

pubdatetime  latitude  longitude  \
0  2019-06-01 00:00:12   36.1202   -86.7534   
1  2019-06-01 00:00:12   36.1630   -86.7765   
2  2019-06-01 00:00:12   36.1202   -86.7533   
3  2019-06-01 00:00:12   36.1201   -86.7530   
4  2019-06-01 00:00:12   36.1622   -86.7806   

                                        sumdid  chargelevel sumdgroup  \
0  Powered93627c35-0f62-5b81-a78d-75a4a92ecf47         90.0   scooter   
1  Powered17715097-e8a0-5494-a5ab-9b625796607d         63.0   scooter   
2  Powerede5cb95ae-b091-5a93-86fa-ededd946d0d7         77.0   scooter   
3  Powered71fa5e4f-1e17-54c4-936d-330df38cc2fa          0.0   scooter   
4  Poweredfa549dd6-40bb-5757-ac87-2c2528f2bc68          2.0   scooter   

   costpermin companyname  
0        0.06        Jump  
1        0.06        Jump  
2        0.06        Jump  
3        0.06        Jump  
4        0.06        Jump

In [19]:
may_scooter_new = may_scooter.drop(columns = 'sumdtype')
may_scooter_new.head()

pubdatetime   latitude  longitude        sumdid  \
0  2019-05-01 00:01:41.247000  36.136822 -86.799877  PoweredLIRL1   
1  2019-05-01 00:01:41.247000  36.191252 -86.772945  PoweredXWRWC   
2  2019-05-01 00:01:41.247000  36.144752 -86.806293  PoweredMEJEH   
3  2019-05-01 00:01:41.247000  36.162056 -86.774688  Powered1A7TC   
4  2019-05-01 00:01:41.247000  36.150973 -86.783109  Powered2TYEF   

   chargelevel sumdgroup  costpermin companyname  
0         93.0   scooter         0.0        Bird  
1         35.0   scooter         0.0        Bird  
2         90.0   scooter         0.0        Bird  
3         88.0   scooter         0.0        Bird  
4         98.0   scooter         0.0        Bird

In [20]:
july_bird = july_scooter_new.loc[july_scooter_new.companyname == 'Bird']

In [21]:
july_bird.head()

pubdatetime   latitude  longitude        sumdid  \
356  2019-07-01 00:01:50.957000  36.163417 -86.779597  PoweredPH3LM   
357  2019-07-01 00:01:50.957000  36.155279 -86.786279  Powered46ZQH   
358  2019-07-01 00:01:50.957000  36.198123 -86.766940  Powered8X3NR   
359  2019-07-01 00:01:50.957000  36.155505 -86.785245  PoweredDBUMF   
360  2019-07-01 00:01:50.957000  36.136868 -86.808101  PoweredP7JJ6   

     chargelevel sumdgroup  costpermin companyname  
356         80.0   scooter        0.15        Bird  
357         48.0   scooter        0.15        Bird  
358         50.0   scooter        0.15        Bird  
359        100.0   scooter        0.15        Bird  
360        100.0   scooter        0.15        Bird

In [22]:
july_gotcha = july_scooter_new.loc[july_scooter_new.companyname == 'Gotcha']

In [23]:
july_gotcha.head()

pubdatetime   latitude  longitude       sumdid  \
625599  2019-07-01 18:38:58.727000  36.121315 -86.770366   Powered399   
625600  2019-07-01 18:38:58.727000  36.121395 -86.770460   Powered349   
625601  2019-07-01 18:38:58.727000  36.121308 -86.770337   Powered365   
625602  2019-07-01 18:38:58.727000  36.121248 -86.770466  Powered2480   
625603  2019-07-01 18:38:58.727000  36.121330 -86.770388   Powered725   

        chargelevel sumdgroup  costpermin companyname  
625599          0.0   Scooter        0.15      Gotcha  
625600         90.0   Scooter        0.15      Gotcha  
625601          0.0   Scooter        0.15      Gotcha  
625602         80.0   Scooter        0.15      Gotcha  
625603          0.0   Scooter        0.15      Gotcha

In [24]:
july_bird['sumdid'].value_counts()

PoweredNTCGG    3682
PoweredFWI2W    3604
PoweredM76B1    3575
PoweredVAW9Y    3539
PoweredTZIK8    3523
                ... 
PoweredPGT79       2
Powered3PD13       2
Powered5WF3M       1
PoweredA9GT8       1
Powered3ZPZ7       1
Name: sumdid, Length: 2583, dtype: int64

In [25]:
total_rides = pd.concat([may_scooter_new,june_scooter_new,july_scooter_new])
total_rides.shape

(73414043, 8)

We see that there are 73,414,043 rides over the past 3 months. 

In [26]:
total_rides.head()

pubdatetime   latitude  longitude        sumdid  \
0  2019-05-01 00:01:41.247000  36.136822 -86.799877  PoweredLIRL1   
1  2019-05-01 00:01:41.247000  36.191252 -86.772945  PoweredXWRWC   
2  2019-05-01 00:01:41.247000  36.144752 -86.806293  PoweredMEJEH   
3  2019-05-01 00:01:41.247000  36.162056 -86.774688  Powered1A7TC   
4  2019-05-01 00:01:41.247000  36.150973 -86.783109  Powered2TYEF   

   chargelevel sumdgroup  costpermin companyname  
0         93.0   scooter         0.0        Bird  
1         35.0   scooter         0.0        Bird  
2         90.0   scooter         0.0        Bird  
3         88.0   scooter         0.0        Bird  
4         98.0   scooter         0.0        Bird

In [27]:
june_1 = june_scooter[(june_scooter['sumdid']=='Powered93627c35-0f62-5b81-a78d-75a4a92ecf47')]  
june_1.head()

pubdatetime  latitude  longitude  \
0      2019-06-01 00:00:12   36.1202   -86.7534   
3932   2019-06-01 00:05:13   36.1202   -86.7534   
7358   2019-06-01 00:10:13   36.1202   -86.7534   
10656  2019-06-01 00:15:12   36.1202   -86.7534   
13854  2019-06-01 00:20:12   36.1202   -86.7534   

                                            sumdid sumdtype  chargelevel  \
0      Powered93627c35-0f62-5b81-a78d-75a4a92ecf47  Powered         90.0   
3932   Powered93627c35-0f62-5b81-a78d-75a4a92ecf47  Powered         90.0   
7358   Powered93627c35-0f62-5b81-a78d-75a4a92ecf47  Powered         90.0   
10656  Powered93627c35-0f62-5b81-a78d-75a4a92ecf47  Powered         90.0   
13854  Powered93627c35-0f62-5b81-a78d-75a4a92ecf47  Powered         90.0   

      sumdgroup  costpermin companyname  
0       scooter        0.06        Jump  
3932    scooter        0.06        Jump  
7358    scooter        0.06        Jump  
10656   scooter        0.06        Jump  
13854   scooter        0.06        Jump

In [28]:
june_precise = june_scooter_new[['latitude','longitude','sumdid','companyname']]
june_precise = june_precise.sort_values(by = 'sumdid')

In [29]:
may_scooter_new = pd.may_scooter_new({'pubdatetime': pd.date_range('2019-06-01 00:00:12', periods=5)})
may_scooter_new.head()

AttributeError: module 'pandas' has no attribute 'may_scooter_new'

In [ ]:
may_scooter_new['new_date'] = [d.date() for d in july_scooter_new['pubdatetime']]
may_scooter_new['new_time'] = [d.time() for d in july_scooter_new['pubdatetime']]